src/vscode-azureappservice.api.d.ts

```diff
@@ -0,0 +1,36 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+export interface AzureAppServiceExtensionApi {
+    apiVersion: string;
+    /**
+     * Creates a web app with the given options.  Empty properties will be their default value or the user will be prompted for it
+     *
+     * @param subscriptionId The subscription id of an Azure subscription
```

Comment:
Most of the params are self-explanatory, so I only included the ones that might be difficult to understand the format.  But should I just explain all the parameters?

Reply:
Soo... did they ask for all of these? When creating an API, it's generally best to only provide as little as possible. Otherwise you're stuck maintaining a bunch of stuff for backwards compatibility that no one's ever using.